### DISCUSSION TITLE
Apply boundary conditions to different parts of stitched mesh

### DISCUSSION THREAD

**Discussion by pratyushpotu** [idaholab/moose#18282] [Q&A Meshing] [answered]
Hi, I am trying to apply boundary conditions to certain boundaries on a mesh comprised of three boxes stitched together horizontally (left, middle, and right). In particular, I'm trying to assign an id to the top part of the left box for instance, but whenever I try applying a boundary condition it applies to the top of the whole mesh instead of just one component. Thus, I was wondering how to apply boundary conditions to different boundaries like the top part of the left box and the bottom part of the right box.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You could try to use
https://mooseframework.inl.gov/source/meshgenerators/BreakBoundaryOnSubdomainGenerator.html
OR
https://mooseframework.inl.gov/source/meshgenerators/BoundingBoxNodeSetGenerator.html

The first will require the boxes to be different blocks. The last one will require you to be a little mindful of floating point evaluations for the node at the corner of two stitched boxes
to split up the top sideset in multiple parts. 

Guillaume

**Reply by pratyushpotu** [reply]
Hi, 

Thanks for your help. I was able to produce the boundaries I wanted, but the meshgenerator system needs a single end point, so when trying to apply boundary conditions to the boundaries, I can only apply a boundary condition to one produced boundary. If I try to apply another boundary condition to a boundary that is not the final generator, I get that the node set id does not exist. How would I go about adding the node set id to a boundary without specifying it to be the final generator?

Thanks,
Pratyush

**Reply by GiudGiud** [reply]
hello

I m not sure I understand the problem. Could you please paste the meshgenerators you are using here?

You should make a linear chain of mesh generators, if that helps

Guillaume

**Reply by GiudGiud** [reply]
oh you are saying the boundaries created by the middle mesh generators are overwritten by the final ones?

So if you add show_info = true to the last mesh generator or if you open the resulting mesh in paraview, you should be able to see the final names of the boundaries after all the generation. You can then use the final names in your boundary conditions

**Reply by pratyushpotu** [reply]
Here are the mesh generators I am using

```
[Mesh]
  [./generated_mesh]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    xmax = 10
    ymin = 0
    ymax = 1
    nx = 100
    ny = 10
  [../]
  [./nodes1]
    type = BoundingBoxNodeSetGenerator
    new_boundary = 'AB'
    input = 'generated_mesh'
    top_right = '2 1 0'
    bottom_left = '2 0 0'
  [../]
  [./nodes2]
    type = BoundingBoxNodeSetGenerator
    new_boundary = 'BC'
    input = 'generated_mesh'
    top_right = '8 1 0'
    bottom_left = '8 0 0'
  [../]
  final_generator = nodes1
  show_info = true
[]
```

And here are the boundary conditions:

```
[BCs]
  [AB_temperature]
    type = DirichletBC
    boundary = 'AB'
    variable = temperature
    value = 500
  []
    [BC_temperature]
    type = DirichletBC
    boundary = 'BC'
    variable = temperature
    value = 500
  []
[]
```

Box A is from x = 0 to x = 2, Box B is from x=2 to x=8, and Box C is from x=8 to x=10. Because I have my final_generator set to nodes1, the AB boundary condition works, but the BC boundary condition does not work. I want to have more boundaries later such as the top of box A and the bottom of box C, but for now I am testing with the interfaces between the three boxes. Therefore, I am wondering how to apply boundary conditions to multiple node sets that are not the default top, bottom etc... . Thanks for your help.

**Reply by GiudGiud** [reply]
ok the problem is that the input of nodes2 needs to be nodes1 not generated_mesh. Otherwise, there are two final meshes and the simulation does not know which one to use. 

or as you did, you selected one as the final generator, and the other one got ignored, so that does not work for you

**Reply by pratyushpotu** [reply]
Oh I see, I thought the input was what part of the mesh the node set was a part of, but its actually just how the mesh generator chains from each generator. Changing the input got me exactly what I needed. Thanks so much for your help.
